implentio/ui-toolkit
# Feat: Create Tooltip Component

## Description
Created the `Tooltip` component and when hovered over the trigger content will display the tooltip. The tooltip includes text and an arrow pointing to trigger content.
Props:

- triggerContent: the component passed in that will be the trigger content. Default value included.
- text: a string of text to be displayed in the tooltip
- side: a string of either 'top', 'bottom', 'right', or 'left' determining which side the tooltip will render

## Link to JIRA ticket
[Link to JIRA ticket](https://implentio.atlassian.net/browse/IUT-26?atlOrigin=eyJpIjoiNTQzMGFlODYwMzg5NDYzOWJkZDJjNTYyNDM1ZDc1YmEiLCJwIjoiaiJ9)

## Applicable Screenshots
 **Note: Icon component is used as an example trigger content to hover and show how tooltip component renders**
<img width="142" alt="Screenshot 2024-02-07 at 4 21 27 PM" src="https://github.com/implentio/ui-toolkit/assets/104450960/8eb97041-2f87-4d71-a28b-e18824c8a4fb">
<img width="148" alt="Screenshot 2024-02-07 at 4 21 44 PM" src="https://github.com/implentio/ui-toolkit/assets/104450960/45a29b50-ad91-42d1-94ff-5c25c9c501fd">
<img width="164" alt="Screenshot 2024-02-07 at 4 22 01 PM" src="https://github.com/implentio/ui-toolkit/assets/104450960/f65d8d40-8a76-4a5b-8a8c-7cd39973057b">
<img width="223" alt="Screenshot 2024-02-07 at 4 22 12 PM" src="https://github.com/implentio/ui-toolkit/assets/104450960/91bf7fa9-c956-452f-b841-1d4eb5fcfb9c">
<img width="227" alt="Screenshot 2024-02-07 at 4 22 47 PM" src="https://github.com/implentio/ui-toolkit/assets/104450960/d14a0af0-665c-447c-9fc4-bbe079755794">
<img width="165" alt="Screenshot 2024-02-07 at 4 24 33 PM" src="https://github.com/implentio/ui-toolkit/assets/104450960/6b0fa353-3008-4f3c-8298-e055ef31e3f8">

 

## Testing Instructions
Outline the steps required to test your changes effectively (Replace the following to what makes sense for your PR):
- Fetch and checkout my branch `git fetch && git checkout feat/IUT-26/Tooltip`
- `npm run storybook`
- Check to see that when hovering over trigger content, the tooltip is displayed. Modify text and side to see if it's displaying properly.

## Checklist:
Check what is applicable by putting an 'x' in between the []:
